velox/experimental/cudf/exec/CudfHashAggregation.cpp

```diff
@@ -507,8 +555,17 @@ auto toIntermediateAggregators(
     auto const inputIndex = aggregationNode.groupingKeys().size() + i;
     auto const kind = aggregate.call->name();
     auto const constant = nullptr;
-    aggregators.push_back(
-        createAggregator(step, kind, inputIndex, constant, isGlobal));
+    const auto originalName = getOriginalName(kind);
+    auto const companionStep = getCompanionStep(kind, step);
+    if (exec::isPartialOutput(companionStep)) {
+      const auto resultType = exec::Aggregate::intermediateType(
+          originalName, aggregate.rawInputTypes);
+      aggregators.push_back(createAggregator(
+          step, kind, inputIndex, constant, isGlobal, resultType));
+    } else {
+      // Final step aggregator will not use the intermediate aggregator.
+      aggregators.push_back(nullptr);
```

Comment:
nit: Is it safe to push nullptr? please make sure it's referenced without nullptr checking.

Reply:
Yes, because intermediate aggregator is only used for step `auto const step = core::AggregationNode::Step::kIntermediate;`, https://github.com/rapidsai/velox/blob/velox-cudf/velox/experimental/cudf/exec/CudfHashAggregation.cpp#L497, but we always generated the intermediate aggregator for all step. In spark, we have mixed step for the aggs, so I insert the not used aggregators as nullptr.